moxin-protocol/src/protocol.rs

```diff
@@ -1,4 +1,42 @@
-use crate::data::Model;
+use crate::data::*;
+use chrono::NaiveDate;
+
+#[derive(Clone, Debug)]
+pub enum ContextOverflowPolicy {
+    StopAtLimit,
+    TruncateMiddle,
+    TruncatePastMessages,
+}
+
+#[derive(Clone, Debug)]
+pub enum GPULayers {
+    Specific(u32),
+    Max,
+}
+
+#[derive(Clone, Debug)]
+pub struct ModelExecutionConfig {
+    pub cpu_threads: u32,
+    pub gpu_layers: GPULayers,
+    pub use_mlock: bool,
+    pub n_batch: u32,
+    pub n_ctx: u32,
+    pub rope_freq_scale: f32,
+    pub rope_freq_base: f32,
+
+    // TBD Not really sure if this is something backend manages or if it is matter of
+    // the client (if it is done by tweaking the JSON payload for the chat completition)
+    pub context_overflow_policy: ContextOverflowPolicy
+}
+
+#[derive(Clone, Debug)]
+pub struct LocalServerConfig {
+    pub port: u16,
+    pub cors: bool,
```

Comment:
Our server is not an HTTP server. Does it need to provide HTTP services?

Reply:
One relevant feature in LM Studio is the ability to spawn a local web server to interact with the model via CURL or scripts.

@juntao What do you think? Maybe it is a topic to cover in the next weekly call, since this is a feature we were not taking about in those calls.